torch/csrc/autograd/profiler_kineto.cpp

```diff
@@ -230,8 +230,17 @@ struct KinetoThreadLocalState : public ProfilerThreadLocalState {
     }
   }
 
+// Assumption: Total threads number will not exceed 2^16-1, and total ops will not exceed 2^48 -1.
+#define GET_FORWARD_THREAD_KEY(tid, seqNr) \
+  (((tid) << 48) | ((seqNr) & (((uint64_t)1 << 48) - 1)))
```

Comment:
How about a union for this:

union {
  uint64_t key;
  struct {
    uint64_t tid : 16;
    uint64_t seqNr : 48;
  } fields;
};

Reply:
I see that this is frowned upon in C++ apparently and not officially supported. Maybe best stick to shifting and masking.